### ISSUE TITLE
3.2-stable with ruby 2.x: ActiveSupport::OrderedHash#to_yaml broken because YAML = Psych

### ISSUE THREAD

**Issue by vindvaki** [rails/rails#22127]
The problem boils down to first few lines of the method
```ruby
def to_yaml(opts = {})
  if YAML.const_defined?(:ENGINE) && !YAML::ENGINE.syck?
    return super
  end
  # ... legacy code for 1.8?
```
in [`activesupport/lib/active_support/ordered_hash.rb`](https://github.com/rails/rails/blob/3-2-stable/activesupport/lib/active_support/ordered_hash.rb#L33)

The reason it is broken is that [in Ruby 2.x, YAML::ENGINE is no longer defined](http://ruby-doc.org/stdlib-2.0.0/libdoc/yaml/rdoc/YAML.html). Instead, we basically have [`YAML = Psych`](https://github.com/ruby/ruby/blob/trunk/lib/yaml.rb#L13). So while the intention of the check is to defer to Psych, it fails to do so when `YAML == Psych`.

**Comment by rafaelfranca** [comment]
Rails 3 is not supported anymore.

**Comment by vindvaki** [comment]
Fair enough. But if I were to submit a pull request, would that have any chance of getting included in 3.2-stable? I ask, because [3.2.22 was claimed to have support for ruby 2.2](http://weblog.rubyonrails.org/2015/6/16/Rails-3-2-22-4-1-11-and-4-2-2-have-been-released-and-more/), and before that these changes were on 3.2-stable.
